core_types/src/eav.rs

```diff
@@ -62,12 +62,13 @@ impl Ord for EntityAttributeValueIndex {
 }
 
 #[derive(Clone, Debug)]
-pub struct IndexQuery {
+pub struct IndexQuery<'a> {
     start: Option<i64>,
     end: Option<i64>,
+    prefixes: Vec<&'a str>,
```

Comment:
Does this really belong in the `IndexQuery`, since it refers to the Attribute? Seems like we should have an `AttributeQuery` as well. Also, I'm going to have to rewrite before merging in #952 because we won't be referring to attributes by raw strings anymore, but enum variants. I don't mind doing that but just pointing out that this will have to change.

Perhaps `AttributeQuery` could just be something like:

```rust
enum AttributeQuery {
    Single(Attribute),
    Predicate(Box<FnOnce(Attribute) -> bool>),
}
```

Reply:
That is definitely a new addition but I think this could be done as a follow up, I kind of don't mind keeping it in the index_query at this point because it is small enough